crates/polars-plan/src/logical_plan/optimizer/mod.rs

```diff
@@ -187,7 +187,14 @@ pub fn optimize(
     lp_top = opt.optimize_loop(&mut rules, expr_arena, lp_arena, lp_top)?;
 
     if members.has_joins_or_unions && members.has_cache {
-        cache_states::set_cache_states(lp_top, lp_arena, expr_arena, scratch, verbose)?;
+        cache_states::set_cache_states(
+            lp_top,
+            lp_arena,
+            expr_arena,
+            scratch,
+            hive_partition_eval,
+            verbose,
+        )?;
```

Comment:
Why does this change require updates to the Rust side? The previous version did not have those changes.